pl_examples/quantization/imagenet.py

```diff
@@ -0,0 +1,374 @@
+# Copyright The PyTorch Lightning team.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""This example is largely adapted from https://github.com/pytorch/examples/blob/master/imagenet/main.py.
+
+Before you can run this example, you will need to download the ImageNet dataset manually from the
+`official website <http://image-net.org/download>`_ and place it into a folder `path/to/imagenet`.
+
+Train on ImageNet with default parameters:
+
+.. code-block: bash
+
+    python imagenet.py --data-path /path/to/imagenet
+
+or show all options you can change:
+
+.. code-block: bash
+
+    python imagenet.py --help
+"""
+import os
+from argparse import ArgumentParser, Namespace
+
+import torch
+import torch.nn.functional as F
+import torch.nn.parallel
+import torch.optim as optim
+import torch.optim.lr_scheduler as lr_scheduler
+import torch.utils.data
+import torch.utils.data.distributed
+import torchvision.datasets as datasets
+import torchvision.models as models
+import torchvision.transforms as transforms
+
+import pytorch_lightning as pl
+from pl_examples import cli_lightning_logo
+from pytorch_lightning.core import LightningModule
+
+
+class ImageNetLightningModel(LightningModule):
```

Comment:
There is already an example with imagenet under `domain_templates`. Let's prevent code duplication.

Reply:
dnoe